src/lib.rs

```diff
@@ -573,12 +577,65 @@ impl DstLayout {
 pub unsafe trait KnownLayout: sealed::KnownLayoutSealed {
     #[doc(hidden)]
     const LAYOUT: DstLayout;
+
+    /// TODO
+    ///
+    /// # Safety
+    ///
+    /// TODO: Mention isize overflow, address space wraparound.
+    #[doc(hidden)]
+    #[inline(always)]
+    fn try_cast_from(bytes: NonNull<[u8]>, cast_type: _CastType) -> Option<(NonNull<Self>, usize)> {
+        let base = bytes.cast::<u8>();
+        let addr = base.as_ptr() as usize;
+        let (elems, split_at) =
+            Self::LAYOUT._validate_cast_and_convert_metadata(addr, bytes.len(), cast_type)?;
+        let slf = match cast_type {
+            _CastType::_Prefix => base,
+            _CastType::_Suffix => {
+                // SAFETY: TODO
+                let base = unsafe { base.as_ptr().add(split_at) };
+                // SAFETY: TODO
+                unsafe { NonNull::new_unchecked(base) }
+            }
+        };
+        Some((Self::raw_from_ptr_len(slf, elems), split_at))
+    }
+
+    /// TODO
+    ///
+    /// # Safety
+    ///
+    /// TODO: Mention isize overflow, address space wraparound.
+    #[doc(hidden)]
+    #[inline(always)]
+    fn try_cast_from_no_prefix_suffix(bytes: NonNull<[u8]>) -> Option<NonNull<Self>> {
```

Comment:
Okay, I've *finally* thought of something better than "no_affix":
```suggestion
    fn try_cast_from_no_excess(bytes: NonNull<[u8]>) -> Option<NonNull<Self>> {
```

Reply:
I still feel like `prefix_suffix`, though a bit more verbose, is the most straightforward and descriptive.

Reply:
Maybe? I'm a little concerned that prefix and suffix are getting a bit overloaded. A prefix cast isn't one with no prefix, and a suffix cast isn't one with no suffix. If anything, this is a cast in which prefix == suffix.

Reply:
Discussed offline; we're going with `try_cast_from_no_leftover`.